cli/src/cli.rs

```diff
@@ -1,99 +1,91 @@
-use std::collections::HashMap;
-use std::io::stderr;
 use std::path::PathBuf;
 
 use clap::Parser;
+use dbg_pls::color;
+use owo_colors::OwoColorize;
 
-use analyzer::importer::ASTImporter;
-use analyzer::name::Name;
-use analyzer::relations::SourceId;
-use analyzer::resolve_all;
-use analyzer::steps::typing::apply_types;
-use analyzer::types::engine::TypedEngine;
-use ast::group::Block;
 use ast::Expr;
-use compiler::compile;
-use context::source::Source;
-use parser::parse;
+use lexer::token::Token;
 
-use crate::report::{display_diagnostic, display_parse_error};
-
-#[derive(Parser)]
+#[derive(Parser, Debug, Clone)]
 #[command(author, version, about, long_about = None)]
 pub struct Cli {
-    /// Defines the source file to parse
+    /// Defines the source file to evaluate
     #[arg(short, long, value_name = "FILE")]
     pub(crate) source: Option<PathBuf>,
+
+    /// Defines the working directory of the cli.
+    #[arg(short, long, value_name = "FILE")]
+    pub(crate) working_dir: Option<PathBuf>,
+
+    /// Activate lexer output visualisation
+    #[arg(short = 'L', long)]
+    pub(crate) lexer: bool,
+
+    /// Activate parser AST visualisation
+    #[arg(short = 'P', long)]
+    pub(crate) parser: bool,
+
+    /// Activate compiler output visualisation
+    #[arg(short = 'B', long)]
+    pub(crate) bytecode: bool,
 }
 
-#[derive(Default)]
-struct RawImporter<'a> {
-    imported_modules: HashMap<Name, Expr<'a>>,
+pub struct Configuration {
```

Comment:
An option to avoid executing the code would be useful.

Reply:
why not indeed